### DISCUSSION TITLE
UX issue: Adding and editing items fail for unclear reasons

### DISCUSSION THREAD

**Discussion by Kovah** [sysadminsmedia/homebox#294] [General]
Adding and editing items in Homebox is a rather unpleasant experience.
Two examples:

Adding new items fails with the message "Couldn't add new item" in the frontend. Checking the Docker logs the following error pops up:
```
ERROR occurred error="ent: constraint failed: constraint failed: FOREIGN KEY constraint failed (787)"
```
Solution: add an item location. It's not clear from the form that the location must be filled. There should at least be proper error handling informing the user that the location is missing.

Editing an item leads to another issue. In the frontend I just get "unable to save item". The logs contain this:
```
body decoding error: json: cannot unmarshal string into Go struct field ItemUpdate.soldPrice of type float64
```
Here again proper handling of the actual form would be more helpful than a generic error message.

**Comment by tankerkiller125** [comment]
For the second one double check that your local browser storage cache is cleared (Ctrl+F5 on Windows and Linux), that issue was fixed in the latest patch release.
